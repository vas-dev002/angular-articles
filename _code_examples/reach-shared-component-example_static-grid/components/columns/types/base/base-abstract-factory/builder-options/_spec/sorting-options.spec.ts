import { SortingOptions } from "..";

declare const expect: jest.Expect;

describe("[Option]: SortingOptions", () => {
	let sortingOptions: SortingOptions;

	beforeEach(() => {
		sortingOptions = new SortingOptions();
	});

	describe("[SortingOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(sortingOptions).toBeTruthy();
			expect(sortingOptions).toMatchSnapshot();
		});
	});

	describe("[Method]: comparator", () => {
		it("should sort correctly if both values are not empty", () => {
			const value1 = "ATest1";
			const value2 = "BTest1";
			const comparator = sortingOptions.comparator;

			expect(comparator(value1, value1)).toEqual(0);
			expect(comparator(value1, value2)).toEqual(-1);
			expect(comparator(value2, value1)).toEqual(1);
		});

		it("should sort correctly if both values are undefined", () => {
			const value1 = null;
			const value2 = null;
			const comparator = sortingOptions.comparator;

			expect(comparator(value1, value1)).toEqual(0);
			expect(comparator(value1, value2)).toEqual(0);
			expect(comparator(value2, value1)).toEqual(0);
		});
	});
});
