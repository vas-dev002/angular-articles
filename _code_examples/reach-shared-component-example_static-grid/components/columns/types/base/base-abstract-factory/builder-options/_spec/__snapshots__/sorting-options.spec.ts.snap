// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Option]: SortingOptions [SortingOptions instantiate] should be successfully created 1`] = `
SortingOptions {
  "comparator": [Function],
  "sortingOrder": Array [
    "asc",
    "desc",
    null,
  ],
}
`;
